### PR TITLE
set correct MTU for VxLAN device

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

Please ensure your pull request adheres to the following guidelines:

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6368)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Review by jrfastab — APPROVED** [review]

File: bpf/init.sh [hunk 1]
```diff
@@ -28,6 +28,8 @@ MTU=$9
 ID_HOST=1
 ID_WORLD=2
 
+TUNNEL_OVERHEAD=50
```
Comment (brb) [inline bpf/init.sh:31]: Isn't the overhead different for IPv6?

**Comment by tgraf** [comment]
The tunnel overhead is accounted for in the routes inside of the pod. Why is this needed? For traffic from host to a remote pod?

File: bpf/init.sh (hunk 1 above)
Comment (tgraf) [reply bpf/init.sh:31]: We always encapsulate over IPv4 so it will be the same.

**Review by vadorovsky — APPROVED** [review]

**Comment by joestringer** [comment]
@tgraf even for traffic from the host to remote pod, I would expect that the MTU of the local route should account for tunnel overhead properly.

**Comment by aanm** [comment]
@brb will test this image as he was seeing retransmission packets while doing benchmark tests

**Comment by tgraf** [comment]
I would expect to see fragmentation if the MTU is incorrect.

**Comment by brb** [comment]
> I would expect to see fragmentation if the MTU is incorrect.

Yes, that's what actually happens. Can be easily verified (`cilium_host` MTU = 8900, `cilium_vxlan` MTU = 1500):

```
root@host1:~# tcpdump -i cilium_vxlan -v -n -e icmp
tcpdump: listening on cilium_vxlan, link-type EN10MB (Ethernet), capture size 262144 bytes
13:57:54.753637 9e:a3:90:5b:32:73 > 1e:64:4f:cd:43:d4, ethertype IPv4 (0x0800), length 8858: (tos 0x0, ttl 64, id 2521, offset 0, flags [+], proto ICMP (1), length 8844)
    10.217.1.62 > 10.217.0.194: ICMP echo request, id 448, seq 1, length 8824
13:57:54.753666 9e:a3:90:5b:32:73 > 1e:64:4f:cd:43:d4, ethertype IPv4 (0x0800), length 218: (tos 0x0, ttl 64, id 2521, offset 8824, flags [none], proto ICMP (1), length 204)
    10.217.1.62 > 10.217.0.194: ip-proto-1
13:57:54.753904 6a:ca:60:28:68:d7 > f2:36:a6:70:13:4d, ethertype IPv4 (0x0800), length 8858: (tos 0x0, ttl 64, id 38047, offset 0, flags [+], proto ICMP (1), length 8844)
    10.217.0.194 > 10.217.1.62: ICMP echo reply, id 448, seq 1, length 8824
13:57:54.753950 6a:ca:60:28:68:d7 > f2:36:a6:70:13:4d, ethertype IPv4 (0x0800), length 218: (tos 0x0, ttl 64, id 38047, offset 8824, flags [none], proto ICMP (1), length 204)
    10.217.0.194 > 10.217.1.62: ip-proto-1

container@host2:# ping -c1 -s 9000 10.217.0.194
PING 10.217.0.194 (10.217.0.194) 9000(9028) bytes of data.
9008 bytes from 10.217.0.194: icmp_seq=1 ttl=63 time=0.755 ms

--- 10.217.0.194 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.755/0.755/0.755/0.000 ms
```

> <..> seeing retransmission packets while doing benchmark tests

The culprit is known: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=9c4c325252c54b34d53b3d0ffd535182b744e03d

I think we can close this PR.

**Pull request closed without merging**